autonomi/src/client/pointer.rs

```diff
@@ -35,20 +39,17 @@ pub enum PointerError {
 
 impl Client {
     /// Get a pointer from the network
-    pub async fn pointer_get(
-        &self,
-        address: PointerAddress,
-    ) -> Result<Pointer, PointerError> {
+    pub async fn pointer_get(&self, address: PointerAddress) -> Result<Pointer, PointerError> {
         let key = NetworkAddress::from_pointer_address(address).to_record_key();
         let record = self.network.get_local_record(&key).await?;
-        
+
         match record {
             Some(record) => {
                 let (_, pointer): (Vec<u8>, Pointer) = rmp_serde::from_slice(&record.value)
                     .map_err(|_| PointerError::Serialization)?;
                 Ok(pointer)
             }
-            None => Err(PointerError::Corrupt),
+            None => Err(PointerError::FailedVerification),
```

Comment:
Does `FailedVerification` cover all possible errors behind `get_local_record`?
I think having at least a String with some information on the actual underlying error would be extremely valuable